### PR TITLE
BE Profile Model

### PR DESCRIPTION
**Pull request by mayanksuchde**
### What this PR does (required):
- Added a Profile model 

### Any information needed to test this feature (required):
- An endpoint for handling requests needs to be added before this feature can be properly tested. This will be worked on next.

### Any issues with the current functionality (optional):
- No issues with the current functionality

### CODE REVIEW

File: server/models/Profile.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const mongoose = require("mongoose");
+const User = require("User");
+const profileSchema = new mongoose.Schema({
+  firstname: {
```
Comment (ezeilo-su) [inline server/models/Profile.js:4]: Prefer camelCase for your variable names

File: server/models/Profile.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    type: String,
+    required: true,
+  },
+  lastname: {
```
Comment (ezeilo-su) [inline server/models/Profile.js:8]: Prefer camelCase for your variable names

**Review by ezeilo-su — COMMENTED** [review]
Your model looks fine. Go ahead and work on adding the endpoints.

File: server/models/Profile.js (hunk 2 above)
Comment (mayanksuchde) [reply server/models/Profile.js]: Thank you, I have made the necessary changes

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  lastName: {
+    type: String,
+    required: true,
+  },
+  dob: {
+    type: Date,
+    required: true,
+  },
+  user: {
```
Comment (rajivtitus) [inline server/models/Profile.js:17]: Wouldn't it be better to add a reference to the User model here instead and maybe call this field `userId`? :)

File: server/models/Profile.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+const mongoose = require("mongoose");
+const userSchema = require("./User");
+const email = User.find_by_id({});
```
Comment (rajivtitus) [inline server/models/Profile.js:3]: What is the purpose of this? Please remove or comment the code with a description :)

File: server/models/Profile.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    required: true,
+  },
+  userPhotoUrl: {
+    type: String,
+  },
+  description: {
+    type: String,
+  },
+  availability: {
+    type: String,
```
Comment (rajivtitus) [inline server/models/Profile.js:28]: How are we planning to store availability here? If we are using 'String', won't this be an array of strings here?

**Review by rajivtitus — COMMENTED** [review]
Good work addressing previous review comments. I've made some new comments, please have a look and address :)

File: server/models/Profile.js (hunk 4 above)
Comment (mayanksuchde) [reply server/models/Profile.js:3]: Done

File: server/models/Profile.js (hunk 3 above)
Comment (mayanksuchde) [reply server/models/Profile.js:17]: Do you mean something like this.
{
    type: mongoose.Schema.Types.ObjectId,
    ref: "user",
    required: true,
}

File: server/models/Profile.js (hunk 5 above)
Comment (mayanksuchde) [reply server/models/Profile.js:28]: Yes it would be an array, Thanks of catching that. Fixed.

File: server/models/Profile.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    required: true,
+  },
+  userPhotoUrl: {
+    type: String,
+  },
+  description: {
+    type: String,
+  },
+  availability: {
+    type: Array,
```
Comment (rajivtitus) [inline server/models/Profile.js:28]: This will be `type: [String]` here :)

**Review by rajivtitus — APPROVED** [review]
LGTM! Please address minor comment before merging. Thanks! :)

File: server/models/Profile.js (hunk 6 above)
Comment (mayanksuchde) [reply server/models/Profile.js:28]: made the change and thanks for the heads up

**Pull request merged**